string filepath
---
string[] not_received